staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -728,6 +728,78 @@ type VirtualMachineInstanceGuestOSInfo struct {
+)
+
+// +k8s:openapi-gen=true
+type VirtualMachineInstanceMigrationSourceState struct {
+	// The source node that the VMI originated on
+	Node string `json:"node,omitempty"`
+	Pod  string `json:"pod,omitempty"`
+	// The Source VirtualMachineInstanceMigration object associated with this migration
+	MigrationUID types.UID `json:"migrationUID,omitempty"`
+	// If the VMI being migrated uses persistent features (backend-storage), its source PVC name is saved here
+	PersistentStatePVCName string `json:"persistentStatePVCName,omitempty"`
+
+	// Node selectors needed by the target to start the receiving pod.
+	NodeSelectors map[string]string `json:"nodeSelectors,omitempty"`
+
+	// The name of the domain on the source libvirt domain
+	DomainName *string `json:"domainName,omitempty"`
+
+	// Namespace used in the name of the source libvirt domain. Can be used to find and modify paths in the domain
+	DomainNamespace *string `json:"domainNamespace,omitempty"`
+
+	// The URL the synchronization controller can connect to synchronize the VMI migration state
+	SynchronizationConnectionURL *string `json:"synchronizationConnectionURL,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+type VirtualMachineInstanceMigrationTargetState struct {
+	// The timestamp at which the target node detects the domain is active
+	NodeDomainReadyTimestamp *metav1.Time `json:"nodeDomainReadyTimestamp,omitempty"`
+	// The Target Node has seen the Domain Start Event
+	NodeDomainDetected bool `json:"nodeDomainDetected,omitempty"`
+	// The address of the target node to use for the migration
+	NodeAddress string `json:"nodeAddress,omitempty"`
+	// The list of ports opened for live migration on the destination node
+	DirectMigrationNodePorts map[string]int `json:"directMigrationNodePorts,omitempty"`
+	// The target node that the VMI is moving to
+	Node string `json:"node,omitempty"`
+	// The target pod that the VMI is moving to
+	Pod string `json:"pod,omitempty"`
+	// The UID of the target attachment pod for hotplug volumes
+	AttachmentPodUID types.UID `json:"attachmentPodUID,omitempty"`
+	// The Target VirtualMachineInstanceMigration object associated with this migration
+	MigrationUID types.UID `json:"migrationUID,omitempty"`
+
+	// If the VMI requires dedicated CPUs, this field will
+	// hold the dedicated CPU set on the target node
+	// +listType=atomic
+	CPUSet []int `json:"cpuSet,omitempty"`
+	// If the VMI requires dedicated CPUs, this field will
+	// hold the numa topology on the target node
+	NodeTopology string `json:"nodeTopology,omitempty"`
+	// If the VMI being migrated uses persistent features (backend-storage), its target PVC name is saved here
+	PersistentStatePVCName string `json:"persistentStatePVCName,omitempty"`
+
+	// The name of the domain on the target libvirt domain
+	DomainName *string `json:"domainName,omitempty"`
+	// Namespace used in the name of the target libvirt domain.
+	DomainNamespace *string `json:"domainNamespace,omitempty"`
```

Comment:
Do we plan on using libvirt namespaces?

Reply:
Yes, so the libvirt domain has the namespace the VM is running in the name. So when we live migrate, the new libvirt domain needs to have a different name since it might be in a different namespace. And that namespace name needs to be transmitted to the source. So that when we call the live migration call, we can rename the target domain correctly.

There are a few more things like that, that gets translated when we migrate.